cmake/ctest_script.cmake

```diff
@@ -396,15 +462,37 @@ endif()
 
 set(ENV{CTEST_USE_LAUNCHERS_DEFAULT} 1)
 
-function(update_gismo updcount)
-  ctest_update(RETURN_VALUE updcount)
-  if(UPDATE_MODULES)
-    execute_process(COMMAND "${CTEST_UPDATE_COMMAND}" "submodule" "update" "--remote"
-      WORKING_DIRECTORY ${gismo_SOURCE_DIR}
-      #RESULT_VARIABLE gresult
-      #OUTPUT_QUIET
-      )
+macro(get_git_status res)
+  execute_process(COMMAND ${CTEST_UPDATE_COMMAND} rev-parse --verify HEAD
+    WORKING_DIRECTORY ${CTEST_SOURCE_DIRECTORY}
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    OUTPUT_VARIABLE gitHash)
+    execute_process(COMMAND ${CTEST_UPDATE_COMMAND} submodule
+      WORKING_DIRECTORY ${CTEST_SOURCE_DIRECTORY}
+      OUTPUT_VARIABLE submoduleHashes)
+    set(${res} " ${gitHash} gismo\n${submoduleHashes}\n")
+endmacro(get_git_status)
+
+function(update_gismo ug_ucount)
+  #this should be last.
+  ctest_update(SOURCE ${CTEST_SOURCE_DIRECTORY} RETURN_VALUE updcount)
```

Comment:
Think this updates submodules too - if the change on HEAD, will be get the update count?

Reply:
This should update the submodules not to HEAD, but detached mode..